src/instruments/src/EFB/Service/TODCalculator.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * A32NX
+ * Copyright (C) 2020-2021 FlyByWire Simulations and its contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {get, orderBy, min, max, inRange, set, last, head} from 'lodash';
+
+type groundSpeed = {from: number, groundSpeed: number};
+type groundSpeedRange= {groundSpeed: number, range: [number, number]};
+
+export default class TODCalculator {
```

Comment:
Do you think you can make this a function component?

Reply:
It's not a component, it's just a calculation class. yes it can be splitted in functions, but can't see any pros of that

Reply:
Ah apologies, I haven't seen a react class in a while and must have mistaken it for one. It's fine as is